restcomm/restcomm.commons/src/main/java/org/restcomm/connect/commons/util/DigestAuthentication.java

```diff
@@ -97,6 +127,28 @@ private static void validate(final String user, final String realm, final String
             throw new NullPointerException("The uri parameter may not be null.");
         } else if (nonce == null) {
             throw new NullPointerException("The nonce parameter may not be null.");
+        } else if (algorithm == null) {
+            throw new NullPointerException("The algorithm parameter may not be null.");
         }
     }
+
+    /**
+     * @param username
+     * @param realm
+     * @param password
+     * @return
+     */
+    public static String HA1(String username, String realm, String password){
+        String algorithm = RestcommConfiguration.getInstance().getMain().getClientAlgorithm();
```

Comment:
could we pass the algorithm as parameter here to reduce dependency to RestCommConfiguration in util class?. This may facilitate a next iteration where algorithm is specified per Client, instead of systemWide setting...

Reply:
removing dependency from RestCommConfiguration in utill class would mean to replicate it in all calling classes. for now its ok, it would make more sense when we wil support multiple algorithms